src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -261,12 +299,21 @@ public final void visitToken(DetailAST blockCommentNode) {
                 result = TREE_CACHE.get().get(treeCacheKey);
             }
             else {
-                result = context.get().parser.parseJavadocAsDetailNode(blockCommentNode);
+                result = context.get().parser
+                        .parseJavadocAsDetailNode(blockCommentNode);
                 TREE_CACHE.get().put(treeCacheKey, result);
             }
 
             if (result.getParseErrorMessage() == null) {
-                processTree(result.getTree());
+                if (doesCheckAcceptJavadocWithNonTightHtml() || !result.isNonTight()) {
+                    processTree(result.getTree());
+                }
+
+                if (violateExecutionOnNonTightHtml && result.isNonTight()) {
+                    log(result.getFirstNonTightHtmlTag().getLine(),
+                            JavadocDetailNodeParser.MSG_UNCLOSED_HTML,
+                            result.getFirstNonTightHtmlTag().getText());
```

Comment:
importing from `org.antlr.v4.runtime` is controlled inside this class and results in an `ant-phase-verify` failure. It also seems sort of correct to not allow importing parser related units in here.

If what I have done is unacceptable too then an alternative would be to use methods like `getFirstNonTightTagLineNumber` and `getFirstNonTightTagText` in `JavadocDetailNodeParser`. 

Yet Another alternative would be to return a `ParseErrorMessage` object itself from `JavadocDetailNodeParser` which can then be used for logging violation but `ParseErrorMessage` seems to be specifically for parsing errors so we can create another similar class like `NonTightHtmlViolationMessage`.